File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int wordCount(string text);
float averageLetters(string text);
float averageSentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float L = averageLetters(text);
    float S = averageSentences(text);
    int index = (int) round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Word count
int wordCount(string text)
{
    int wordCount = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            wordCount += 1;
        }
    }
    return wordCount;
}

// Average letters per 100 words
float averageLetters(string text)
{
    int letterCount = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letterCount += 1;
        }
    }
    float averageLetters = ((float) letterCount * 100) / (float) wordCount(text);
    return averageLetters;
}

// Average sentences per 100 words
float averageSentences(string text)
{
    int sentenceCount = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (strchr(".!?", text[i]) != NULL)
        {
            sentenceCount += 1;
        }
    }
    float averageSentences = ((float) sentenceCount * 100) / (float) wordCount(text);
    return averageSentences;
}

File: sentimental-readability/readability.py
# Readability Python script

from cs50 import get_string


def main():
    text = get_string("Text: ")
    L = averageLetters(text)
    S = averageSentences(text)
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(index))


def averageLetters(text):
    letterCount = 0
    for i in range(len(text)):
        if (text[i].isalpha()):
            letterCount += 1
    averageLetters = (letterCount * 100) / wordCount(text)
    return averageLetters


def averageSentences(text):
    sentenceCount = 0
    sentenceEnders = [".", "!", "?"]
    for i in range(len(text)):
        if any(sentenceEnder in sentenceEnders for sentenceEnder in text[i]):
            sentenceCount += 1
    averageSentences = (sentenceCount * 100) / wordCount(text)
    return averageSentences


def wordCount(text):
    wordCount = 1
    for i in range(len(text)):
        if text[i] == ' ':
            wordCount += 1
    return wordCount


main()
